examples/http/main.go

```diff
@@ -18,40 +18,45 @@ import (
 
 func main() {
 	var (
-		serverMode = flag.Bool("server", false, "Run as server")
-		clientMode = flag.Bool("client", false, "Run as client")
-		host       = flag.String("host", "localhost", "Host to connect to or listen on")
-		port       = flag.String("port", "8080", "Port to connect to or listen on")
+		host = flag.String("host", "localhost", "Host to connect to or listen on")
+		port = flag.String("port", "8080", "Port to connect to or listen on")
 	)
 
 	flag.Usage = func() {
-		fmt.Fprintf(os.Stderr, "Usage: %s [options]\n\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "Usage: %s <client|server> [options]\n\n", os.Args[0])
 		fmt.Fprintf(os.Stderr, "This program demonstrates MCP over HTTP using the streamable transport.\n")
 		fmt.Fprintf(os.Stderr, "It can run as either a server or client.\n\n")
 		fmt.Fprintf(os.Stderr, "Options:\n")
 		flag.PrintDefaults()
 		fmt.Fprintf(os.Stderr, "\nExamples:\n")
-		fmt.Fprintf(os.Stderr, "  Run as server:  %s -server\n", os.Args[0])
-		fmt.Fprintf(os.Stderr, "  Run as client:  %s -client\n", os.Args[0])
-		fmt.Fprintf(os.Stderr, "  Custom host/port: %s -server -host 0.0.0.0 -port 9090\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "  Run as server:  %s server\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "  Run as client:  %s client\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "  Custom host/port: %s server -host 0.0.0.0 -port 9090\n", os.Args[0])
 		os.Exit(1)
 	}
 
-	flag.Parse()
-
-	if (*serverMode && *clientMode) || (!*serverMode && !*clientMode) {
-		fmt.Fprintf(os.Stderr, "Error: Must specify exactly one of -server or -client\n\n")
+	// Check if we have at least one argument
+	if len(os.Args) < 2 {
```

Comment:
Call flag.Parse first, then access args as flag.Arg(0), etc. Use flag.NArg() for the count.

Reply:
How about we simplify it even mode and make it  `./http <client|server> [http://localhost:9000]` 

This would be good because with this, I can test the server mode on a VPS with TLS cert issued to Caddy and proxying requests to:

./http server http://localhost:8000

and then I can connect to it with the client:

./http client https://mymcp.koszek.com:443

Reply:
@jba fixed this part.